modules/gapi/src/backends/cpu/gcpuvideo.cpp

```diff
@@ -107,13 +107,66 @@ GAPI_OCV_KERNEL_ST(GCPUBackgroundSubtractor,
     }
 };
 
+GAPI_OCV_KERNEL_ST(GCPUKalmanFilter, cv::gapi::video::GKalmanFilter, cv::KalmanFilter)
+{
+    static void setup(const cv::GMatDesc&, const cv::GOpaqueDesc&,
+                      const cv::GMatDesc&, const cv::gapi::video::KalmanParams& kfParams,
+                      std::shared_ptr<cv::KalmanFilter> &state, const cv::GCompileArgs&)
+    {
+
+       // auto kfParams = cv::gapi::getCompileArg<cv::gapi::video::KalmanParams>(compileArgs)
+         //               .value_or(cv::gapi::video::KalmanParams{});
+
+        GAPI_Assert(kfParams.type == CV_32F || kfParams.type == CV_64F);
+        state = std::make_shared<cv::KalmanFilter>(kfParams.dpDim, kfParams.mpDim, kfParams.ctrlDim, kfParams.type);
+
+        // initial state
+        kfParams.statePre.copyTo(state->statePre);
+        kfParams.errorCovPre.copyTo(state->errorCovPre);
```

Comment:
why use `copyTo`? cv::Mats can be shared I believe.

Reply:
Changed.